sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,780 @@
+
+    return_model_to_cpu : bool, default = False
+        After fitting and training, will return the `momentfm` model to the cpu.
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "handles-missing-data": False,
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "python_dependencies": [
+            "torch",
+            "tqdm",
+            "huggingface-hub",
+            # "momentfm",
+            "accelerate",
+        ],
+        "capability:global_forecasting": True,
+        "python_version": ">= 3.10",
+        "capability:insample": False,
+        "capability:pred_int:insample": False,
+        "capability:pred_int": False,
+    }
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path="AutonLab/MOMENT-1-large",
+        freeze_encoder=True,
+        freeze_embedder=True,
+        freeze_head=False,
+        dropout=0.1,
+        head_dropout=0.1,
+        seq_len=512,
+        batch_size=32,
+        eval_batch_size=32,
+        epochs=1,
+        max_lr=1e-4,
+        device="gpu",
+        pct_start=0.3,
+        max_norm=5.0,
+        train_val_split=0.4,
+        transformer_backbone="google/flan-t5-large",
+        config=None,
+        return_model_to_cpu=False,
+    ):
+        super().__init__()
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.freeze_encoder = freeze_encoder
+        self.freeze_embedder = freeze_embedder
+        self.freeze_head = freeze_head
+        self.dropout = dropout
+        self.head_dropout = head_dropout
+        self.seq_len = seq_len
+        self.batch_size = batch_size
+        self.eval_batch_size = eval_batch_size
+        self.epochs = epochs
+        self.max_lr = max_lr
+        self.device = device
+        self.pct_start = pct_start
+        self.max_norm = max_norm
+        self.train_val_split = train_val_split
+        self.transformer_backbone = transformer_backbone
+        self.config = config
+        self._config = config if config is not None else {}
+        self.criterion = MSELoss()
+        self._moment_seq_len = 512
+        self.return_model_to_cpu = return_model_to_cpu
+
+    def _fit(self, fh, y, X=None):
+        """Assumes y is a single or multivariate time series."""
+        from torch.optim import Adam
+        from torch.optim.lr_scheduler import OneCycleLR
+        from torch.utils.data import DataLoader
+
+        # initialize accelerator
+        accelerator = Accelerator()
+        # keep a copy of y in case y is None in predict
+        self._y = y
+        self._y_index = self._y.index
+        self._y_cols = self._y.columns
+        self._y_shape = self._y.values.shape
+        self._pretrained_model_name_or_path = (
```

Comment:
For all of the following lines, would it be easier to read if you would write something like:

`self._pretrained_model_name_or_path = self._config.get("_pretrained_model_name_or_path", self. _pretrained_model_name_or_path)`